### PR TITLE
Add pow() function for numeric

### LINKED ISSUE #963: Document power(numeric, numeric)
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/10028

### Which part(s) of the docs might be affected or should be updated? And how?

I see we have pow (dp, dp) but not pow(numeric, numeric). 

### Reference

https://www.postgresql.org/docs/15/functions-math.html

### PR DESCRIPTION
**Pull request by bernscode**
Add pow() function for numeric.

## Info

- **Description**

Add pow() function for numeric.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

https://github.com/risingwavelabs/risingwave/pull/10028

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/963

## For reviewers

- **Preview**

https://pr-980.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-function-mathematical/

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ title: Mathematical functions and operators
...
 | ln ( *double_precision_input* ) → *double_precision_output* <br /> ln ( *numeric_input* ) → *numeric_output*| Returns the natural logarithmic value of the input. | ln(10) → 2.302585092994046 |
 | log10 ( *double_precision_input* ) → *double_precision_output* <br /> log10 ( *numeric_input* ) → *numeric_output*| Returns the log base 10 value of the input value. log() can also be used and accepts the same input types. | log10(25) → 1.3979400086720377 |
-| pow ( *x_double_precision*, *y_double_precision* ) → *double_precision_output* <br /> power ( *x_double_precision*, *y_double_precision* ) → *double_precision_output* | Returns *x_double_precision* raised to the power of *y_double_precision*. | pow(2.0, 3.0) → 8 <br /> power(2.0, 3.0) → 8|
+| pow ( *x_double_precision*, *y_double_precision* ) → *double_precision* <br /> power ( *x_double_precision*, *y_double_precision* ) → *double_precision* <br /> pow ( *x_numeric*, *y_numeric* ) → *numeric* | Returns *x_double_precision* or *x_numeric* raised to the power of *y_double_precision* or *y_numeric*. | pow(2.0, 3.0) → 8 <br /> power(2.0, 3.0) → 8|
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-mathematical.md]: Maybe follow the convention of ceil/ceiling and log10/log to only list one name in the leftmost column and mention the alternative spelling in description?

**Review by xiangjinwu — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-mathematical.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-mathematical.md]: @xiangjinwu Good point! I've made the updates on the latest commit. Thanks!

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 2]
```diff
@@ -26,12 +26,12 @@ title: Mathematical functions and operators
...
 | ln ( *double_precision_input* ) → *double_precision_output* <br /> ln ( *numeric_input* ) → *numeric_output*| Returns the natural logarithmic value of the input. | ln(10) → 2.302585092994046 |
 | log10 ( *double_precision_input* ) → *double_precision_output* <br /> log10 ( *numeric_input* ) → *numeric_output*| Returns the log base 10 value of the input value. log() can also be used and accepts the same input types. | log10(25) → 1.3979400086720377 |
-| pow ( *x_double_precision*, *y_double_precision* ) → *double_precision_output* <br /> power ( *x_double_precision*, *y_double_precision* ) → *double_precision_output* | Returns *x_double_precision* raised to the power of *y_double_precision*. | pow(2.0, 3.0) → 8 <br /> power(2.0, 3.0) → 8|
+| pow ( *x_double_precision*, *y_double_precision* ) → *double_precision* <br /> pow ( *x_numeric*, *y_numeric* ) → *numeric* | Returns *x_double_precision* or *x_numeric* raised to the power of *y_double_precision* or *y_numeric*. power() can also be used as an alias for pow() when using the double precision data type. | pow(2.0, 3.0) → 8 <br /> power(2.0, 3.0) → 8|
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-mathematical.md]: > when using the double precision data type

Actually both names work for both data types. Maybe I should have made it clear in the implementation PR 😂

Comment (bernscode) [reply docs/sql/functions-operators/sql-function-mathematical.md]: @xiangjinwu No worries at all. I've removed that part in the latest commit. Thanks!

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**